### PR TITLE
chore: use `jest-matcher-utils` in `@types/jest`

### PR DESCRIPTION
**Pull request by SimenB**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

In an ongoing quest to make #44365 mergable, let's try to load more types from Jest itself. This time `this.utils` inside custom matchers.

I was unable to test this, just getting `Error: ENOENT: no such file or directory, open '../notNeededPackages.json'` when attempting to run `npm install && npm test jest` 🤷 

EDIT: `npm run test-all` works

### CODE REVIEW

**Comment by SimenB** [comment]
CI failure: https://github.com/microsoft/DefinitelyTyped-tools/pull/399

File: types/jest/jest-tests.ts [hunk 1]
```diff
@@ -858,7 +858,7 @@ expect.extend({
 
         const receivedPrinted: string = this.utils.printReceived({});
 
-        const printedWithType: string = this.utils.printWithType('name', {}, (value: {}) => '');
+        const printedWithType: string = this.utils.printWithType('name', {}, (value) => '');
```
Comment (SimenB) [inline types/jest/jest-tests.ts:861]: type in jest is `unknown`, not `any`

Comment (SimenB) [reply types/jest/jest-tests.ts:861]: fixed in https://github.com/facebook/jest/pull/12405, but I think this change is fine anyways

**Review by jakebailey — APPROVED** [review]

**Comment by SimenB** [comment]
Ready to merge

**Pull request merged**